staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -62,6 +62,42 @@ var (
 		},
 		[]string{"resource"},
 	)
+	watchersCounter = metrics.NewCounterVec(
```

Comment:
What is the point of adding these metrics at this level.
Based on flags passed to apiserver and default configuration, we know which resources are using cacher and which aren't. So keeping track of these metrics at this level isn't what we should do imho.
They should be handled at the apiserver level (staging/src/k8s.io/apiserver/pkg/endpoint/...)

Reply:
It's reasonable to move this metric to `staging/src/k8s.io/apiserver/pkg/endpoint/metrics`